### ISSUE TITLE
Investigate the key bindings for executing code in various scenarios

### ISSUE THREAD

**Issue by petetronic** [posit-dev/positron#1461]
Jupyter Notebook:
Shift + Enter -> runs code, advances cursor focus to next cell
Cmd + Enter -> runs code, keeps cursor focused where it is
Option + Enter -> runs code, inserts new cell
Run selected text (No key binding, but a menu option. This is not common.)

Quarto will match this Jupyter Notebook behavior, but we also need an execute line based keyboard shortcut. TBD: Can we steal Option + Enter?
 
VSCode Jupyter:
Shift + Enter -> runs cell, advances cursor focus to next cell
Cmd + Enter -> currently does nothing (but should do the same as ctrl)
Ctrl + Enter -> runs code cell, keeps focus where it is

RStudio Quarto Visual Editor:
Cmd + Enter -> runs the line, goes to next line in same cell
Cmd + Shift + Enter -> runs the entire cell, focus stays in the same cell
Option + Cmd + N -> runs the next chunk

RStudio Editor:
Cmd + Shift + Enter: -> Runs the whole file

VSCode Python Jupyter-Like Cell:
Shift + Enter -> Runs cell, sends output to Interactive Terminal, creates new cell and advances focus.
Cmd + Enter -> Nothing, just inserts new line
Ctrl + Enter -> Runs cell, sends output to Interactive Terminal, leaves focus in current cell.

VSCode Python Extension:
Shift + Enter -> keyboard has been taken over, but should have been Run line/selection in Terminal

Positron Editor:
Cmd + Enter -> sends line/selection to Console
Shift + Enter -> sends line 
Cmd + Shift + Enter -> source file to Terminal

Positron Python Jupyter-like Cell:
Cmd + Enter -> Sends line/selection to Console
Ctrl + Enter -> Runs cell an sends result to Interactive Window
Shift + Enter -> Sends line/selection to Console
Cmd + Shift + Enter -> Sources python file in Console (Positron)

**Comment by GitHunter0** [comment]
> But we could take over Cmd + Enter to just mean execute the current line and advance the code line but stay in same cell?

That would be great. Currently in `positron`, `Ctrl+Enter` in `.R` files runs the current code block / selection and **advances** to the next. I cannot find a way to keep the cursor/focus where it is...

**Comment by juliasilge** [comment]
@GitHunter0 I believe you may be looking for what is requested in https://github.com/posit-dev/positron/issues/2778. Take a look and see if you have additional info to add!

**Comment by GitHunter0** [comment]
@juliasilge , thank you Julia, that's exactly what I wanted.

**Comment by jaredlander** [comment]
Another option from RStudio that I love is alt+enter which runs the current line but does not advance the cursor to the next line. Trying to add that myself in my keybindings but can't figure out the right command. I'm having trouble figuring out if it is in workbench,action.* or positron.* or something else. I tried looking up where ctrl+enter is defined to learn from that, but can't quite figure out where that is specified for `langId == 'r'` as I don't see it in the default keybindings.

**Comment by jaredlander** [comment]
And...I just realized I opened #2778 and I never saw @juliasilge mentioning that it is hard to untangle the commands right now. But I guess this shows how much I hope it is possible one day.

**Comment by juliasilge** [comment]
Yep @jaredlander unfortunately we don't have those two pieces (execute, then advance) separated out right now so that they can be bound to keybindings separately. Thank you for this feedback about its importance, though! 🙌

**Comment by rodrigosf672** [comment]
@petetronic Is this still relevant in your opinion? Is the goal here just to verify the keybindings or do we actually want to add/modify any keybindings?

**Comment by seeM** [comment]
The last outstanding piece was addressed in #2778. We can track remaining work separately (as in #8918).
